### PR TITLE
Refactor the code for school library

### PR DESCRIPTION
**Pull request by Bellagirl-maker**
Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
Follow SOLID principles.
Pay special attention to the "Single-responsibility Principle".
Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: library.rb [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+      student = Student.new(nil, age, name, parent_permission: true)
+      @people << student
+    when 'N'
+      student = Student.new(nil, age, name, parent_permission: false)
+      @people << student
+    else
+      puts 'Invalid input!'
+    end
+    puts 'Person Created Successfully!'
+  end
```
Comment (Gambit142) [inline library.rb]: - [ ] In the `create_student` method, I would advise you to follow the `SOLID` principle while refactoring the code. All codes that ask the users for inputs or validate the user's input should be put in a different **_method, class or module._** The `create_student` method should be solely responsible for creating the student. 👍 💻  

#### Hint: 
Codes like this that ask users for input should be put in a separate **_method, class method or module method:_**

```ruby
print 'Student age: '
age = gets.chomp
print 'Student name: '
name = gets.chomp
```

Codes like this that validate users' inputs should be put in a separate **_method, class method or module method:_**

```ruby
 case permission
 when 'Y'
 parent_permission = true
```

Codes like this that create a new instance of the Student class should be put in a separate **_method, class method or module method:_**

```ruby
student = Student.new(nil, age, name, parent_permission: true)
@people << student
```

### N.B: Kindly implement these changes for the following methods:
- `create_teacher`
- `create_book`
- `create_rental`

File: library.rb [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    puts 'Rental created successfully'
+  end
+
+  def list_rentals
+    print 'Person ID: '
+    id = gets.chomp.to_i
+    @rentals.each do |rent|
+      puts "Date: #{rent.date}, Book: \"#{rent.book.title}\" by #{rent.book.author}" if rent.person.id == id
+    end
+  end
```
Comment (Gambit142) [inline library.rb]: - [ ] In the `list_rentals` method, I would advise you to follow the `SOLID` principle while refactoring the code. All codes that ask the users for inputs or validate the user's input should be put in a different **_method, class or module._** The `list_rental` method should be solely responsible for listing rentals that belong to a particular person according to the video sample provided. You can refer to this [comment](https://github.com/Bellagirl-maker/oop-school-library/pull/5#discussion_r1251128021) for better understanding 👍 💻  

#### N.B: Please ensure that the application works properly after refactoring the codes. 😎 🛸

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Bellagirl-maker @aima98 👋, 

Good job so far!

## Highlights
- All linters checks passed ✔️ 
- Good commit messages ✔️ 
- Pull request summary is descriptive ✔️ 

You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

You can also consider the following:

N/A

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: library.rb (hunk 1 above)
Comment (Bellagirl-maker) [reply library.rb]: @Gambit142 done

File: library.rb (hunk 2 above)
Comment (Bellagirl-maker) [reply library.rb]: @Gambit142 done

**Review by Davidosky007 — APPROVED** [review]
Hi @Bellagirl-maker 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
* SOLID principle followed 👍 
* Well written PR 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**